src/main/resources/META-INF/rewrite/spring-boot-20.yml

```diff
@@ -28,6 +28,11 @@ description: >
 tags:
   - spring
   - boot
+preconditions:
+  - org.openrewrite.java.dependencies.DependencyInsight:
+      groupIdPattern: org.springframework
+      artifactIdPattern: spring-core
+      version: 4.x
 recipeList:
```

Comment:
I'd like to move this precondition to a narrower subrecipe -- right now, it's fully halting the "chain" of each upgrade recipe calling the previous, It also means that recipes such `SpringBoot2BestPractices` are no longer run for projects starting on newer Spring versions, which is perhaps unexpected and I'd say undesirable.

Maybe I move the precondition to `MigrateHibernateConstraintsToJavax`? Any concerns @BoykoAlex ?

Reply:
I think that still makes the unit test that I've added to pass. Perhaps we can narrow down the check as well to `jakarta.validation-api` transient dependency presence.